### DISCUSSION TITLE
Supabase auth

### DISCUSSION THREAD

**Discussion by gerardocrr** [supabase-community/supabase-csharp#164] [Q&A]
I have a blazor wasm project, I am trying to authenticate a user, the sign in works correctly, but it does not save the session, that is to say every time I want to retrieve the user information I have to log in again.

If I log in and click on the button to see the session information it returns the data correctly, but if I log in and close the project and run it again it returns the session as null when it should have the session saved, can someone help me with this?

This is the code I have

```
private async Task SignIn()
{
    var user = await supabase.Auth.SignIn(email, password);
}

private void GetSession()
{
    var user = supabase.Auth.CurrentSession;

    string metadataJson = JsonConvert.SerializeObject(user, Formatting.Indented);
    Console.WriteLine(metadataJson);    
}
```

**Comment by wiverson** [comment]
Where are you persisting the session? There’s a callback interface to
implement persistence. If you are trying to persist on the client you might
want to look at the JS client instead. If you are trying to do this in the
server you might want to combine setting the JWT in a cookie with the
stateless C# APIs…

--------------------------------------------------------
https://changenode.com/
***@***.***

**Reply by gerardocrr** [reply]
I don't know how to make the session persist, I thought it was saved automatically like in javascript, I checked the [documentation](https://supabase.com/docs/reference/csharp/introduction) and I can't find how to make the session persist.

**Reply by Phantom-KNA** [reply]
> I don't know how to make the session persist, I thought it was saved automatically like in javascript, I checked the [documentation](https://supabase.com/docs/reference/csharp/introduction) and I can't find how to make the session persist. 

Check this:
https://gist.github.com/Phantom-KNA/0eabbbe52076370489d0ecbf73f0a6c6

This example are in the documentation

**Reply by gerardocrr** [reply]
It didn't work either, I think blazor handles the cache differently as it gave me some errors with that and I changed it to localstorage but it doesn't work either, I have been following this blazor wasm [example ](https://github.com/supabase-community/supabase-csharp/tree/master/Examples/BlazorWebAssemblySupabaseTemplate) as it is and it doesn't work either, I also cloned the repository and ran it on my machine without modifying the code and it doesn't work either, it logs in correctly but the session doesn't persist.
